File: context/poc/agent/agent/tools/email.py
import os
from datetime import datetime, timezone

from agents import function_tool


@function_tool
def send_email(email_address: str, subject: str, body: str) -> str:
    """Send an email to the specified address."""
    try:
        os.makedirs("/data/emails", exist_ok=True)
        timestamp = datetime.now(timezone.utc).strftime("%Y%m%d_%H%M%S")
        filename = f"/data/emails/{timestamp}.txt"
        with open(filename, "w") as f:
            f.write(f"To: {email_address}\n")
            f.write(f"Subject: {subject}\n")
            f.write(f"Date: {datetime.now(timezone.utc).isoformat()}\n")
            f.write(f"\n{body}\n")
        return f"Email sent to {email_address} (saved to {filename})"
    except Exception as e:
        return f"Error sending email: {e}"

File: example_agents/adk_js/src/tools/email.ts
import fs from 'node:fs';
import path from 'node:path';
import { FunctionTool } from '@google/adk';
import { z } from 'zod';

const BASE_DIR = process.cwd();

export const sendEmail = new FunctionTool({
  name: 'send_email',
  description: 'Send an email to the specified address.',
  parameters: z.object({
    email_address: z.string().describe('Recipient email address'),
    subject: z.string().describe('Email subject'),
    body: z.string().describe('Email body'),
  }),
  execute({ email_address, subject, body }) {
    try {
      const emailsDir = path.join(BASE_DIR, '.data', 'emails');
      fs.mkdirSync(emailsDir, { recursive: true });
      const now = new Date();
      const ts = now.toISOString().replace(/[-:T]/g, '').slice(0, 15);
      const filename = path.join(emailsDir, `${ts}.txt`);
      const content = `To: ${email_address}\nSubject: ${subject}\nDate: ${now.toISOString()}\n\n${body}\n`;
      fs.writeFileSync(filename, content, 'utf-8');
      return { sent_to: email_address, saved_to: filename };
    } catch (e: unknown) {
      return { error: (e as Error).message };
    }
  },
});
